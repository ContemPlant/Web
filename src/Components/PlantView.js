import React from 'react'
import { Popup, Grid } from 'semantic-ui-react'
import IndividualImage from './IndividualImage'
import HealthIcon from './HealthIcon'
import ArduLoader from './ArduLoader'

class PlantView extends React.Component {

    constructor(props) {
        super(props)
        const health = (this.props.plant.plantStates[0] && this.props.plant.plantStates[0].health || 0) * 100

        this.state = {
            plantName: this.props.plant.name,
            plantId: this.props.plant.id,
            health: health,
            tempOpt: this.props.plant.temperature_opt,
            humOpt: this.props.plant.humidity_opt,
            radOpt: this.props.plant.radiation_opt,
            loudOpt: this.props.plant.loudness_opt,
            history: this.props.history
        }
    }

    isLoaded() {
        const { loadedPlantId } = sessionStorage
        const { plantId } = this.props

        return loadedPlantId && loadedPlantId === plantId
            ? true
            : false
    }

    onDetail(e) {
        sessionStorage.health = this.state.health
        sessionStorage.plantId = this.state.plantId
        sessionStorage.plantName = this.state.plantName
        sessionStorage.tempOpt = this.state.tempOpt
        sessionStorage.humOpt = this.state.humOpt
        sessionStorage.radOpt = this.state.radOpt
        sessionStorage.loudOpt = this.state.loudOpt
        this.state.history.push("/detail")
    }

    render() {

        return (
            <div>
                <Grid>
                    <Grid.Column width={4} verticalAlign="middle">
                        <IndividualImage id={this.state.plantId} />
                    </Grid.Column>
                    <Grid.Column width={4} verticalAlign="middle">
                        <Popup
                            trigger={<a onClick={(e) => this.onDetail(e)}><h2>{this.props.plant.name}</h2></a>}
                            content="Show plant details"
                            basic
                        />
                    </Grid.Column>
                    <Grid.Column width={4} verticalAlign="middle">
                        <HealthIcon health={this.state.health} />
                    </Grid.Column>
                    <Grid.Column width={4} verticalAlign="middle">
                        <ArduLoader health={this.state.health} plantId={this.state.plantId} plantName={this.state.plantName} />
                    </Grid.Column>
                </Grid>

            </div>
        )
    }
}


export default PlantView
