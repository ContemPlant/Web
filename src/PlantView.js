import React from 'react';
import { Link } from "react-router-dom";
import { Icon, Segment, Button, Popup, Grid, Image } from 'semantic-ui-react';
import IndividualImage from './IndividualImage';
import HealthIcon from './HealthIcon';
import ArduLoader from './ArduLoader';


class PlantView extends React.Component {

	constructor(props) {
            super(props);
            this.state = {
              jwt: sessionStorage.jwt,
              plantName: this.props.plant.name,
              plantId: this.props.plant.id,
              health: (this.props.plant.plantStates[0] && this.props.plant.plantStates[0].health || 0)*100,
							loaded: this.isLoaded()
			};
			console.log(this.props.plant.id)
	}

				isLoaded() {
					if (sessionStorage.loadedPlantId !== null && sessionStorage.loadedPlantId === this.props.plantId) {
						return true
					} else {
						return false
					}
				}

	onDetail(e) {
		sessionStorage.health = this.state.health,
		sessionStorage.plantId = this.state.plantId,
		sessionStorage.plantName = this.state.plantName,
		this.props.history.push("/detail")
	}


    render() {

        return (
        	<div>
          <Grid>
            <Grid.Column width={4} verticalAlign="middle">
              <IndividualImage id={this.state.plantId} />
            </Grid.Column>
            <Grid.Column width={4} verticalAlign="middle">
						<Popup
								trigger={ <a onClick={ (e) => this.onDetail(e) }><h2>{this.props.plant.name}</h2></a> }
								content="Show plant details"
								basic
						/>
            </Grid.Column>
            <Grid.Column width={4} verticalAlign="middle">
              <HealthIcon health={this.state.health} />
            </Grid.Column>
            <Grid.Column width={4} verticalAlign="middle">
              <ArduLoader health={this.state.health} plantId={this.state.plantId} plantName={this.state.plantName}/>
            </Grid.Column>
          </Grid>

			    </div>
        )
    }
}

export default PlantView;
