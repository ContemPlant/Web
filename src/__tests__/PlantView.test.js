import React from 'react'
import PlantView from '../Components/PlantView'
import { shallow, mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

configure({ adapter: new Adapter() })


describe('PlantView', () => {
    const plant = {
        id: 'plant1',
        name: 'fanky',
        temperature_opt: 20,
        humidity_opt: 10,
        radiation_opt: 30,
        loudness_opt: 300,
        plantStates: [{
            healht: 1, environment: 0, size: 1337
        }]
    }

    test('Renders correctly', () => {
        const tree = shallow(
            <PlantView plant={plant} />
        )
        expect(tree).toMatchSnapshot()
    })
})