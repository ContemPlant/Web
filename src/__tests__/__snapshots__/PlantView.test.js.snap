// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantView Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlantView
    plant={
        Object {
            "humidity_opt": 10,
            "id": "plant1",
            "loudness_opt": 300,
            "name": "fanky",
            "plantStates": Array [
              Object {
                "environment": 0,
                "healht": 1,
                "size": 1337,
              },
            ],
            "radiation_opt": 30,
            "temperature_opt": 20,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Grid>
        <GridColumn
                verticalAlign="middle"
                width={4}
        >
                <IndividualImage
                        id="plant1"
                />
        </GridColumn>
        <GridColumn
                verticalAlign="middle"
                width={4}
        >
                <Popup
                        basic={true}
                        content="Show plant details"
                        keepInViewPort={true}
                        on="hover"
                        position="top left"
                        trigger={
                                <a
                                        onClick={[Function]}
                                >
                                        <h2>
                                                fanky
                                        </h2>
                                </a>
                        }
                />
        </GridColumn>
        <GridColumn
                verticalAlign="middle"
                width={4}
        >
                <HealthIcon
                        health={0}
                />
        </GridColumn>
        <GridColumn
                verticalAlign="middle"
                width={4}
        >
                <ArduLoader
                        health={0}
                        plantId="plant1"
                        plantName="fanky"
                />
        </GridColumn>
</Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridColumn
            verticalAlign="middle"
            width={4}
>
            <IndividualImage
                        id="plant1"
            />
</GridColumn>,
          <GridColumn
            verticalAlign="middle"
            width={4}
>
            <Popup
                        basic={true}
                        content="Show plant details"
                        keepInViewPort={true}
                        on="hover"
                        position="top left"
                        trigger={
                                    <a
                                                onClick={[Function]}
                                    >
                                                <h2>
                                                            fanky
                                                </h2>
                                    </a>
                        }
            />
</GridColumn>,
          <GridColumn
            verticalAlign="middle"
            width={4}
>
            <HealthIcon
                        health={0}
            />
</GridColumn>,
          <GridColumn
            verticalAlign="middle"
            width={4}
>
            <ArduLoader
                        health={0}
                        plantId="plant1"
                        plantName="fanky"
            />
</GridColumn>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <IndividualImage
              id="plant1"
/>,
            "verticalAlign": "middle",
            "width": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "plant1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Popup
              basic={true}
              content="Show plant details"
              keepInViewPort={true}
              on="hover"
              position="top left"
              trigger={
                            <a
                                          onClick={[Function]}
                            >
                                          <h2>
                                                        fanky
                                          </h2>
                            </a>
              }
/>,
            "verticalAlign": "middle",
            "width": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "basic": true,
              "content": "Show plant details",
              "keepInViewPort": true,
              "on": "hover",
              "position": "top left",
              "trigger": <a
                onClick={[Function]}
>
                <h2>
                                fanky
                </h2>
</a>,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <HealthIcon
              health={0}
/>,
            "verticalAlign": "middle",
            "width": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "health": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ArduLoader
              health={0}
              plantId="plant1"
              plantName="fanky"
/>,
            "verticalAlign": "middle",
            "width": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "health": 0,
              "plantId": "plant1",
              "plantName": "fanky",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Grid>
          <GridColumn
                    verticalAlign="middle"
                    width={4}
          >
                    <IndividualImage
                              id="plant1"
                    />
          </GridColumn>
          <GridColumn
                    verticalAlign="middle"
                    width={4}
          >
                    <Popup
                              basic={true}
                              content="Show plant details"
                              keepInViewPort={true}
                              on="hover"
                              position="top left"
                              trigger={
                                        <a
                                                  onClick={[Function]}
                                        >
                                                  <h2>
                                                            fanky
                                                  </h2>
                                        </a>
                              }
                    />
          </GridColumn>
          <GridColumn
                    verticalAlign="middle"
                    width={4}
          >
                    <HealthIcon
                              health={0}
                    />
          </GridColumn>
          <GridColumn
                    verticalAlign="middle"
                    width={4}
          >
                    <ArduLoader
                              health={0}
                              plantId="plant1"
                              plantName="fanky"
                    />
          </GridColumn>
</Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GridColumn
              verticalAlign="middle"
              width={4}
>
              <IndividualImage
                            id="plant1"
              />
</GridColumn>,
            <GridColumn
              verticalAlign="middle"
              width={4}
>
              <Popup
                            basic={true}
                            content="Show plant details"
                            keepInViewPort={true}
                            on="hover"
                            position="top left"
                            trigger={
                                          <a
                                                        onClick={[Function]}
                                          >
                                                        <h2>
                                                                      fanky
                                                        </h2>
                                          </a>
                            }
              />
</GridColumn>,
            <GridColumn
              verticalAlign="middle"
              width={4}
>
              <HealthIcon
                            health={0}
              />
</GridColumn>,
            <GridColumn
              verticalAlign="middle"
              width={4}
>
              <ArduLoader
                            health={0}
                            plantId="plant1"
                            plantName="fanky"
              />
</GridColumn>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <IndividualImage
                id="plant1"
/>,
              "verticalAlign": "middle",
              "width": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "plant1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Popup
                basic={true}
                content="Show plant details"
                keepInViewPort={true}
                on="hover"
                position="top left"
                trigger={
                                <a
                                                onClick={[Function]}
                                >
                                                <h2>
                                                                fanky
                                                </h2>
                                </a>
                }
/>,
              "verticalAlign": "middle",
              "width": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "basic": true,
                "content": "Show plant details",
                "keepInViewPort": true,
                "on": "hover",
                "position": "top left",
                "trigger": <a
                  onClick={[Function]}
>
                  <h2>
                                    fanky
                  </h2>
</a>,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <HealthIcon
                health={0}
/>,
              "verticalAlign": "middle",
              "width": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "health": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ArduLoader
                health={0}
                plantId="plant1"
                plantName="fanky"
/>,
              "verticalAlign": "middle",
              "width": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "health": 0,
                "plantId": "plant1",
                "plantName": "fanky",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
